src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,664 @@
+					__( 'Cancel activation request' )
+				);
+
+				// Use `array_merge` to make sure the action is added as the 1st item in the array.
+				$actions = array_merge( array( 'cancel-activation' => $cancel_activation ), $actions );
+			}
+		}
+		return $actions;
+	}
+
+	/**
+	 * Get a plugin's parents.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @param string $plugin_file Path to the plugin file relative to the plugins' directory.
+	 *
+	 * @return array[] Returns an array of parent plugins.
+	 */
+	public function get_parents( $plugin_file ) {
+		if ( ! isset( $this->dependencies_parents[ $plugin_file ] ) ) {
+			return array();
+		}
+		return $this->dependencies_parents[ $plugin_file ];
+	}
+
+	/**
+	 * Add dependencies info in plugins.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @param string               $plugin_file Path to the plugin file relative to the plugins' directory.
+	 * @param array<string, mixed> $plugin_data An array of plugin data.
+	 *
+	 * @return void
+	 */
+	public function after_plugin_row( $plugin_file, $plugin_data ) {
+
+		$pending_activation     = in_array( $plugin_file, $this->get_plugins_to_activate(), true );
+		$dependencies           = $this->get_plugin_dependencies( $plugin_file );
+		$has_dependencies       = ! empty( $dependencies );
+		$in_circular_dependency = $this->in_circular_dependency( $plugin_file );
+		$is_plugin_active       = is_plugin_active( $plugin_file );
+		$parent_dependencies    = $this->get_parents( $plugin_file );
+
+		// Add extra info to dependencies.
+		// If the plugin is active and a dependency for another plugin, it cannot be deactivated.
+		if ( ! empty( $parent_dependencies ) && $is_plugin_active ) {
+			$parents_names = array();
+			foreach ( $parent_dependencies as $parent ) {
+				$parents_names[] = get_plugin_data( WP_PLUGIN_DIR . '/' . $parent )['Name'];
+			}
+
+			$notice_contents = sprintf(
+				/* translators: %1$s: plugin name. %2$s: Parent plugin name. */
+				esc_html__( 'Plugin %1$s cannot be deactivated because it is a dependency for the "%2$s" plugin.' ),
+				esc_html( $plugin_data['Name'] ),
+				esc_html( $parents_names[0] )
+			);
+			if ( 1 < count( $parents_names ) ) {
+				$notice_contents = sprintf(
+					/* translators: %1$s: plugin name. %2$s: Parent plugin names, comma-separated. */
+					esc_html__( 'Plugin %1$s cannot be deactivated because it is a dependency for the following plugins: %2$s.' ),
+					esc_html( $plugin_data['Name'] ),
+					esc_html( implode( ', ', $parents_names ) )
+				);
+			}
+
+			$this->inline_plugin_row_notice( $notice_contents, 'info', true );
+		}
+
+		// Early return if the plugin doesn't have dependencies.
+		if ( ! $has_dependencies ) {
+			return;
+		}
+
+		// Add a notice if the plugin is pending activation.
+		if ( $pending_activation ) {
+			if ( $in_circular_dependency ) {
+				$this->inline_plugin_row_notice(
+					sprintf(
+					/* translators: %s: plugin name. */
```

Comment:
I think these translator comments were previously aligned correctly, and are now missing a tab as of 6f27089.

Same for two instances below. The rest are still aligned correctly :)

Reply:
Thanks Sergey! Fixed :+1: